lib/SILOptimizer/Transforms/ArrayElementValuePropagation.cpp

```diff
@@ -53,18 +57,25 @@ class ArrayAllocation {
 
   // The calls to Array get_element that use this array allocation.
   llvm::SmallSetVector<ApplyInst *, 16> GetElementCalls;
+  llvm::SmallVector<ApplyInst *, 4> AppendContentsOfCalls;
   llvm::DenseMap<uint64_t, SILValue> ElementValueMap;
+  llvm::SmallVector<SILValue, 4> ElementValueVector;
 
   // Array get_element calls and their matching array element value for later
   // replacement.
-  llvm::SmallVectorImpl<std::pair<ApplyInst *, SILValue>> &ReplacementMap;
+  llvm::SmallVectorImpl<std::pair<ApplyInst *, SILValue>> &GetElementReplacementMap;
+
+  // Array append_contentsOf calls and their matching array values for later
+  // replacement.
+  llvm::SmallVectorImpl<std::pair<ApplyInst *, llvm::SmallVector<SILValue, 4> > > &AppendContentsOfReplacementMap;
 
   ArrayAllocation(
       ApplyInst *AI,
-      llvm::SmallVectorImpl<std::pair<ApplyInst *, SILValue>> &Replacements)
-      : Alloc(AI), ReplacementMap(Replacements) {}
+      llvm::SmallVectorImpl<std::pair<ApplyInst *, SILValue>> &GetElementReplacements,
+      llvm::SmallVectorImpl<std::pair<ApplyInst *, llvm::SmallVector<SILValue, 4> > > &AppendContentsOfReplacements)
```

Comment:
I tried to make

`llvm::SmallVectorImpl<std::pair<ApplyInst *, llvm::SmallVector<SILValue, 4> > >`

instead be

`llvm::SmallVectorImpl<std::pair<ApplyInst *, llvm::SmallVectorImpl<SILValue> > >`

but it fails to compile because it can't find a conversion from `std::pair<ApplyInst *, llvm::SmallVector<SILValue, 4> >` to `std::pair<ApplyInst *, llvm::SmallVectorImpl<SILValue> >`. I don't know C++ well enough to understand what's wrong.